WcaOnRails/app/controllers/registrations_controller.rb

```diff
@@ -508,23 +508,29 @@ def stripe_webhook
 
       stored_intent = stored_transaction.stripe_payment_intent
 
-      stored_intent.update_status_and_charges(stripe_intent, audit_event) do |charge_transaction|
+      stored_intent.update_status_and_charges(stripe_intent, audit_event, audit_event.created_at_remote) do |charge_transaction|
```

Comment:
Why do you need to give the method `audit_event.created_at_remote` if it already has audit_event, can't you just use action.source.created_at_remote in the update_status_and_charge method?

Reply:
Because in general, it cannot be determined that the source timestamp always comes from `action_source`. For example, if the servers don't crash and the user is successfully redirected to the `redirect_url` in Stripe, then the `action_source` is set to be the `User` object of the user who paid. That user does not have a `created_at_remote` timestamp. So we cannot make any generalised assumptions here